packages/text-bitmap/src/BitmapFont.ts

```diff
@@ -226,4 +236,298 @@ export class BitmapFont
     {
         return available;
     }
+
+    /**
+     * Generates a bitmap-font for the given style and character set. This does not support
+     * kernings yet.
+     *
+     * @param {PIXI.IBitmapFontFactoryOptions} options - includes all `ITextStyle` properties.
+     * @param {string|string[]} options.chars - characters included in the font set. You can also
+     *      use ranges. For example, `[['a', 'z'], ['A', 'Z'], "!@#$%^&*()~`{}[]"]. Don't forget to
+     *      include spaces ' ' in your character set!
+     * @param {number} [options.resolution=1] - optional resolution to render the glyphs at
+     * @return {PIXI.BitmapFont} font generated by style options.
+     * @static
+     * @example
+     * const font = PIXI.BitmapFont.from({
+     *     chars: [['a', 'z'], ['A', 'Z'], "!#$%^&*()~`{}[]"]
+     *     name: "TitleFont", // optional
+     *     fontFamily: "Arial",
+     *     fontSize: 12,
+     *     strokeThickness: 2,
+     *     fill: "purple"
+     * });
+     *
+     * const text = new PIXI.BitmapText("This is an example using Pixi's BitmapText+Font", font);
+     *
+     * const title = new PIXI.BitmapText("This is the title", { font: "TitleFont" });
+     */
+    public static from(options: IBitmapFontFactoryOptions): BitmapFont
```

Comment:
I think it'd be nice if the options for the bitmap font, such as chars, name etc, were separate from the text style options, such as FontFamily, fontSize etc. Then it makes it easier to transition from PIXI.Text to PIXI.BitmapText

Reply:
I was thinking about this too. Then we could pass in a shared TextStyle object instead of converting a raw object.